File: AwesomeArchComponent/app/src/main/java/com/akexorcist/repositoryarchcomponents/db/DatabaseManager.java
package com.akexorcist.repositoryarchcomponents.db;

import android.app.Application;
import android.arch.persistence.room.Room;

import com.akexorcist.repositoryarchcomponents.db.hero.HeroDatabase;

/**
 * Created by Akexorcist on 10/6/2017 AD.
 */

public class DatabaseManager {
    private static HeroDatabase heroDatabase;

    public static void init(Application app) {
        heroDatabase = Room.databaseBuilder(app, HeroDatabase.class, "hero.db").build();
    }

    public static HeroDatabase getHeroDatabase() {
        return heroDatabase;
    }
}

File: AwesomeArchComponent/app/src/main/java/com/akexorcist/repositoryarchcomponents/db/DatabaseManager.kt
package com.akexorcist.repositoryarchcomponents.db

import android.app.Application
import android.arch.persistence.room.Room

import com.akexorcist.repositoryarchcomponents.hero.HeroDatabase


/**
 * Created by Akexorcist on 10/6/2017 AD.
 */

object DatabaseManager {
    var heroDatabase: HeroDatabase? = null
        private set

    fun init(app: Application) {
        heroDatabase = Room.databaseBuilder(app, HeroDatabase::class.java, "hero.db").build()
    }
}
